src/components/editor/FileSelector.tsx

```diff
@@ -1,51 +1,97 @@
-import { List, ListItemButton, ListItemText } from '@mui/material';
+import { ListItemText } from '@mui/material';
+import {
+    DataGrid,
+    GridColDef,
+    GridRenderCellParams,
+    GridSelectionModel,
+} from '@mui/x-data-grid';
 import { LiveSpecQuery } from 'components/capture/details';
 import { EditorStoreState, useZustandStore } from 'components/editor/Store';
 import { DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { useEffect, useState } from 'react';
 import { FormattedMessage } from 'react-intl';
 
-function EditorFileSelector() {
-    const currentCatalog = useZustandStore<
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>,
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>['currentCatalog']
-    >((state) => state.currentCatalog);
+const initialState = {
+    columns: {
+        columnVisibilityModel: {
+            spec_type: false,
+        },
+    },
+};
+
+const getRowId = (spec: any) => {
+    let newSelectionModel;
+
+    if (spec.id) {
+        newSelectionModel = spec.id;
+    } else if (spec.draft_id) {
+        newSelectionModel = spec.draft_id;
+    }
+
+    return `${newSelectionModel}-${spec.catalog_name}`;
```

Comment:
Supporting both LiveSpec and DraftSpec is kinda annoying cause DraftSpec does not have a unique id. I assume this is probably good enough but we shall see as we test.